src/main/java/reactor/netty/http/server/WebsocketServerOperations.java

```diff
@@ -206,9 +225,15 @@ void sendCloseNow(@Nullable CloseWebSocketFrame frame, ChannelFutureListener lis
 			return;
 		}
 		if (CLOSE_SENT.getAndSet(this, 1) == 0) {
-			ChannelFuture f = channel().writeAndFlush(
-					frame == null ? new CloseWebSocketFrame() : frame);
-			f.addListener(listener);
+			if (frame != null) {
+				onCloseState.onNext(new WebSocketCloseStatus(frame.statusCode(), frame.reasonText()));
+				channel().writeAndFlush(frame)
+				         .addListener(ChannelFutureListener.CLOSE);
+			} else {
+				onCloseState.onNext(new WebSocketCloseStatus(-1, ""));
+				channel().writeAndFlush(new CloseWebSocketFrame())
+				         .addListener(ChannelFutureListener.CLOSE);
```

Comment:
add `listener` and not ChannelFutureListener.CLOSE

see the previous code `f.addListener(listener);`